activiti-core/activiti-engine/src/main/java/org/activiti/engine/impl/cfg/RollbackEnvironment.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2010-2020 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.activiti.engine.impl.cfg;
+
+public class RollbackEnvironment {
+
+    public static final String ACTIVITI_DEPLOY_AFTER_ROLLBACK_VARIABLE_NAME = "ACTIVITI_DEPLOY_AFTER_ROLLBACK";
+
+    public boolean isRollbackEnabled() {
+        return Boolean.parseBoolean(System.getenv(ACTIVITI_DEPLOY_AFTER_ROLLBACK_VARIABLE_NAME));
```

Comment:
I think this should be a primitive boolean (default false) with set/get. Reading the env var and setting the value should be from a spring property somewhere outside the core engine code (maybe activiti-cloud).

Reply:
It makes sense, thanks for the feeback.
We can do it from the `SpringProcessEngineConfiguration`, from there we can inject the property via `@Value` annotation into the `ProcessEngineConfiguration`

Reply:
Yeah, `@Value` can be injected here [ApplicationUpgradeContextAutoConfiguration](https://github.com/Activiti/Activiti/blob/4255-support-rollback-engine-level/activiti-core-common/activiti-spring-project/src/main/java/org/activiti/core/common/spring/project/conf/ApplicationUpgradeContextAutoConfiguration.java#L38) to [ApplicationUpgradeContextService](https://github.com/Activiti/Activiti/blob/4255-support-rollback-engine-level/activiti-core-common/activiti-spring-project/src/main/java/org/activiti/core/common/spring/project/ApplicationUpgradeContextService.java) which `SpringProcessEngineConfiguration` can use to set the flag. Seems to me ApplicationUpgradeContextService is the right place for it as we are doing an upgrade here.

Reply:
`SpringProcessEngineConfiguration ` can read the property from `ApplicationUpgradeContextService` [here](https://github.com/Activiti/Activiti/blob/4255-support-rollback-engine-level/activiti-core/activiti-spring/src/main/java/org/activiti/spring/SpringProcessEngineConfiguration.java#L69)